.github/workflows/continuous-integration.yml

```diff
@@ -54,7 +54,15 @@ jobs:
 
     - name: Run all checks
       run: |
-        nix flake check --print-build-logs
+        # this is just `nix flake check --print-build-logs`
+        # everything else is to buffer the output for speed
+        (
+          set +e
+          nix flake check --print-build-logs >out 2>&1
+          errcode=$?
+          cat out
+          exit $errcode
+        )
```

Comment:
It might be a naive question, but what's the point of using an intermediate file? My (ignorant) understanding is that `stderr` is unbuffered, and `stdout` is buffered. If redirecting a unbuffered stream to a buffered stream makes it buffered, would

```bash
nix flake check --print-build-logs 2>&1
```

be sufficient?